PerfTools/AllocMonitorPreload/src/memory_proxies.cc

```diff
@@ -0,0 +1,419 @@
+#include <memory>
+#include <cassert>
+#include <atomic>
+#include <cstddef>
+#include <malloc.h>
+
+#include "PerfTools/AllocMonitor/interface/AllocMonitorRegistry.h"
+
+#include <dlfcn.h>  // dlsym
+
+#if !defined(__x86_64__) && !defined(__i386__)
+#define USE_LOCAL_MALLOC
+#endif
+
+namespace {
+  std::atomic<bool>& alloc_monitor_running_state() {
+    static std::atomic<bool> s_state = false;
+    return s_state;
+  }
+
+  template <typename T>
+  T get(const char* iName) {
+    void* original = dlsym(RTLD_NEXT, iName);
+    assert(original);
+    return reinterpret_cast<T>(original);
+  }
+
+#ifdef USE_LOCAL_MALLOC
+  // this is a very simple-minded allocator used for any allocations
+  // before we've finished our setup.  In particular, this avoids a
+  // chicken/egg problem if dlsym() allocates any memory.
+  constexpr auto max_align = alignof(std::max_align_t);
+  alignas(max_align) char tmpbuff[131072];
```

Comment:
Could we add some comment on the size of the buffer (128 kB), like is it based on some estimation, or a guess that seems to work? (thinking a future reader that wonders why the size was set to 128 kB)

Reply:
The libstdc++ exception handling reserves an emergency buffer, which on ARM64 defaults to 72704 bytes.  That triggers a call to dlsym() to load malloc(), which allocates another 32 bytes for the error handler buffer.  So I took 72736 and rounded up to the next power of 2 to be on the safe side.